### PR TITLE
Implement shield blocking

### PR DESCRIPTION
**Pull request by OnlyBMan**

### CODE REVIEW

**Comment by DoctorMacc** [comment]
Something I forgot to do - Redned might ask for the shield ID to be a constant in BlockTranslator (see the LlamaEntity class for an example).

Also - if anyone would like to double-check the inventory checking logic and see if it can be simplified, that would be great.

**Comment by LegacyGamerHD** [comment]
> Something I forgot to do - Redned might ask for the shield ID to be a constant in BlockTranslator (see the LlamaEntity class for an example).
> 
> Also - if anyone would like to double-check the inventory checking logic and see if it can be simplified, that would be great.

already checking it out doesnt work with oldcombat macanics blocking is very wanky
i get easely hit by skelleton and zombies creapers are fine

**Comment by DoctorMacc** [comment]
Later I can test with OCM but that is definitely less of a priority if vanilla mechanics still work.

**Comment by LegacyGamerHD** [comment]
do blocking issues with skelitons and zombies

**Comment by LegacyGamerHD** [comment]
?

File: connector/src/main/java/org/geysermc/connector/entity/Entity.java [hunk 1]
```diff
@@ -196,6 +204,28 @@ public void updateBedrockMetadata(EntityMetadata entityMetadata, GeyserSession s
                     metadata.getFlags().setFlag(EntityFlag.SPRINTING, (xd & 0x08) == 0x08);
                     metadata.getFlags().setFlag(EntityFlag.SWIMMING, (xd & 0x10) == 0x10);
                     metadata.getFlags().setFlag(EntityFlag.GLIDING, (xd & 0x80) == 0x80);
+
+                    // Shield code
+                    if (session.getPlayerEntity().getEntityId() == entityId && metadata.getFlags().getFlag(EntityFlag.SNEAKING)) {
+                        if ((session.getInventory().getItemInHand() != null && session.getInventory().getItemInHand().getId() == 829) ||
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/entity/Entity.java:210]: Please make this ID (829) a constant ID in `BlockTranslator`.

File: connector/src/main/java/org/geysermc/connector/entity/Entity.java [hunk 2]
```diff
@@ -196,6 +204,28 @@ public void updateBedrockMetadata(EntityMetadata entityMetadata, GeyserSession s
...
+                                (session.getInventoryCache().getPlayerInventory().getItem(45) != null && session.getInventoryCache().getPlayerInventory().getItem(45).getId() == 829)) {
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/entity/Entity.java:211]: Same as above.

File: connector/src/main/java/org/geysermc/connector/entity/Entity.java [hunk 3]
```diff
@@ -196,6 +204,28 @@ public void updateBedrockMetadata(EntityMetadata entityMetadata, GeyserSession s
...
+                            ClientPlayerUseItemPacket useItemPacket;
+                            metadata.getFlags().setFlag(EntityFlag.BLOCKING, true);
+                            if (session.getInventory().getItemInHand() != null && session.getInventory().getItemInHand().getId() == 829) {
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/entity/Entity.java:214]: Same as above.

File: connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInteractTranslator.java [hunk 4]
```diff
@@ -46,6 +47,9 @@ public void translate(InteractPacket packet, GeyserSession session) {
 
         switch (packet.getAction()) {
             case INTERACT:
+                if(session.getInventory().getItem(session.getInventory().getHeldItemSlot() + 36).getId() == 829) {
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInteractTranslator.java:50]: Space after `if` and make this ID (829) a constant ID in BlockTranslator.

File: connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java [hunk 5]
```diff
@@ -75,6 +75,9 @@ public void translate(InventoryTransactionPacket packet, GeyserSession session)
                         session.getDownstream().getSession().send(blockPacket);
                         break;
                     case 1:
+                        if (session.getInventory().getItem(session.getInventory().getHeldItemSlot() + 36).getId() == 829) {
```
Comment (RednedEpic) [inline connector/src/main/java/org/geysermc/connector/network/translators/bedrock/BedrockInventoryTransactionTranslator.java:78]: Make this ID (829) a constant ID in BlockTranslator.

**Pull request merged**